lib/foreman_remote_execution_core/script_runner.rb

```diff
@@ -279,5 +282,17 @@ def check_expecting_disconnect
         @expecting_disconnect = true
       end
     end
+
+    def available_authentication_methods
+      methods = %w(pubkey) # Always use pubkey auth as fallback
+      if settings.kerberos_auth
```

Comment:
Do you run smart_proxy_dynflow_core as a separate process or as a part of smart proxy? From what I remember I expected `settings` to be an `OpenStruct` whereas in your case it is a `Hash`. Will investigate tomorrow